test/inductor/test_flex_attention.py

```diff
@@ -614,32 +698,37 @@ def score_mod_2(score, b, h, m, n):
 
         composed_score_mod = _compose(score_mod_1, score_mod_2)
 
-        self.run_test(composed_score_mod, dtype)
+        self.run_test(composed_score_mod, dtype, decoding=decoding)
 
     @supported_platform
     @common_utils.parametrize("dtype", test_dtypes)
-    def test_captured_buffers(self, dtype: torch.dtype):
-        head_offset = torch.rand(H, device="cuda", dtype=dtype)
+    @common_utils.parametrize("decoding", [False, True])
+    def test_captured_buffers(self, dtype: torch.dtype, decoding: bool):
+        head_offset = torch.rand(Hkv if decoding else H, device="cuda", dtype=dtype)
 
         def score_mod(score, b, h, m, n):
             return score + head_offset[h]
 
-        self.run_test(score_mod, dtype)
+        self.run_test(score_mod, dtype, decoding=decoding)
 
     @supported_platform
     @common_utils.parametrize("dtype", test_dtypes)
-    def test_captured_buffers_all_dims(self, dtype: torch.dtype):
-        head_scale = torch.randn(H, device="cuda")
+    @common_utils.parametrize("decoding", [False, True])
+    def test_captured_buffers_all_dims(self, dtype: torch.dtype, decoding):
+        head_scale = torch.randn(Hkv if decoding else H, device="cuda")
         batch_scale = torch.randn(B, device="cuda")
-        tok_scale = torch.randn(S, device="cuda")
+        kv_scale = torch.randn(S, device="cuda")
+        q_scale = torch.randn(Hq // Hkv if decoding else S, device="cuda")
+        if q_scale.shape[-1] < 16:
+            q_scale = torch.nn.functional.pad(q_scale, (0, 16 - Hq // Hkv))
```

Comment:
If users would need to pad the buffers they are going to capture, maybe we should handle this during the lowering? this feels hard to to use

Reply:
The ultimate solution is adding load mask to modification{{}} generated code. Like this: 
https://github.com/joydddd/pytorch/pull/2/files# But we have some performance problems.  

Just a temp work around before we figure that out.